src/libstore/builtins/buildenv.cc

```diff
@@ -13,6 +13,30 @@ struct State
     unsigned long symlinks = 0;
 };
 
+/**
+ * This exception is private and only used within buildenv.cc.
+ * Its purpose is to pass-through contextual error information.
+ * It inherits from std::exception instead of nix::Error to
+ * avoid having to render pretty errors.
+ */
+class BuildEnvFileConflictError : public std::exception
```

Comment:
Why is this a subclass of `std::exception` instead of `Error`? In most of the Nix code base we catch `Error`, not `std::exception`.

Reply:
See https://github.com/NixOS/nix/pull/7788#issuecomment-1438696402